explorer/src/test/java/com/scalar/client/tool/explorer/ExplorerTest.java

```diff
@@ -44,179 +42,150 @@
+    when(contractExecutionResultOK.getResult()).thenReturn(Optional.of(expected));
     when(clientService.executeContract(eq(HOLDER_ID + "GET"), any(JsonObject.class)))
-        .thenReturn(
-            contractExecutionResponseCertificateNotFound,
-            contractExecutionResponseContractNotFound,
-            contractExecutionResponseOK);
+        .thenReturn(contractExecutionResultOK);
 
     // Act
-    JsonObject result = explorer.get(ASSET_ID);
+    JsonObject actual = explorer.get(ASSET_ID);
 
     // Assert
-    verify(clientService, Mockito.times(3))
-        .executeContract(eq(HOLDER_ID + "GET"), any(JsonObject.class));
-    assertThat(result.getString("id")).isEqualTo("bar");
+    verify(clientService).executeContract(eq(HOLDER_ID + "GET"), any(JsonObject.class));
+    assertThat(actual).isEqualTo(expected);
   }
 
   @Test
   public void get_FailedToRegisterCertificate_ShouldThrowExplorerException() {
     // Arrange
-    when(clientService.executeContract(eq(HOLDER_ID + "GET"), any(JsonObject.class)))
-        .thenReturn(contractExecutionResponseCertificateNotFound);
-    when(clientService.registerCertificate()).thenReturn(ledgerServiceResponseCertificateNotFound);
-
-    // Act-assert
-    assertThatThrownBy(() -> explorer.get(ASSET_ID))
-        .isInstanceOf(ExplorerException.class)
-        .hasMessage("403 certificate not found");
-  }
-
-  @Test
-  public void get_CertificateNotFoundTwice_ShouldThrowExplorerException() {
-    // Arrange
-    when(clientService.executeContract(eq(HOLDER_ID + "GET"), any(JsonObject.class)))
-        .thenReturn(
-            contractExecutionResponseCertificateNotFound,
-            contractExecutionResponseCertificateNotFound);
+    when(clientService.executeContract(eq( HOLDER_ID + "GET"), any(JsonObject.class)))
+        .thenThrow(clientExceptionCertificateNotFound);
+    doNothing().when(clientService).registerCertificate();
 
     // Act-assert
     assertThatThrownBy(() -> explorer.get(ASSET_ID))
         .isInstanceOf(ExplorerException.class)
-        .hasMessage("403 certificate not found");
+        .hasMessage("403 Certificate not found");
   }
 
   @Test
-  public void get_ContractNotFoundTwice_ShouldThrowExplorerException() {
+  public void get_CertificateNotFound_ShouldThrowExplorerException() {
     // Arrange
     when(clientService.executeContract(eq(HOLDER_ID + "GET"), any(JsonObject.class)))
-        .thenReturn(
-            contractExecutionResponseContractNotFound, contractExecutionResponseContractNotFound);
+        .thenThrow(clientExceptionCertificateNotFound);
 
     // Act-assert
     assertThatThrownBy(() -> explorer.get(ASSET_ID))
         .isInstanceOf(ExplorerException.class)
-        .hasMessage("404 contract not found");
+        .hasMessage("403 Certificate not found");
   }
 
   @Test
-  public void scan_ThrowExceptionOnlyOneTime_ShouldReturnSuccess() {
+  public void scan_ShouldReturnSuccess() {
```

Comment:
```suggestion
  public void scan_ProperArgumentGiven_ShouldReturnSuccess() {
```
please follow the format: methodName_condition_expectedBehavior.

Reply:
Thank you for the review comment.
Review changes updated in 877c01f